packages/core/src/State.ts

```diff
@@ -55,8 +54,11 @@ export class State<
 > {
   public value: StateValue;
   public context: TContext;
-  public historyValue?: HistoryValue | undefined;
   public history?: State<TContext, TEvent, TStateSchema>;
+  public historyMap: Record<
```

Comment:
This is currently only set inside `resolveTransition`, but previously it was possible to rehydrate `historyValue` through, at least, both `State.from` and `State.create` calls. This seems like an important missing piece for successful state rehydration.